plugin: expose subflow error to plugins

marco-a-itl opened the pull request:
Add error parameter to subflow_closed callback to allow plugins to identify the reason for subflow closure, such as network errors.

The error value corresponds to the `MPTCP_ATTR_ERROR` attribute from the kernel, which equals `sk_err` from `struct sock`. If the error attribute is not present in the Netlink message, it defaults to 0.

Fixes #330

## Changes
- Updated `mptcpd_plugin_ops.subflow_closed` signature to include `uint8_t error` parameter
- Modified all plugins and tests to use the new signature
- Added test validation for the error parameter

marco-a-itl committed:
plugin: expose subflow error to plugins

Add error parameter to subflow_closed callback to allow plugins
to identify the reason for subflow closure, such as network errors.

The error value corresponds to the MPTCP_ATTR_ERROR attribute from
the kernel, which equals sk_err from struct sock. If the error
attribute is not present in the netlink message, it defaults to 0.

Fixes #330

matttbe commented on tests/lib/test-plugin.h:
```diff
@@ -160,13 +161,15 @@ static mptcpd_aid_t   const test_raddr_id_2    = 0x45;
 static bool           const test_backup_4      = true;
 static bool           const test_server_side_4 = false;
 static bool           const test_deny_join_id0_4 = false;
+static uint8_t        const test_error_4       = 0;
```
small detail: maybe good to put a non 0 value for one of them, just to check something different, no? That will not change much, but just by principle, just in case?

matttbe commented on include/mptcpd/plugin.h:
```diff
@@ -247,13 +247,15 @@ struct mptcpd_plugin_ops
          * @param[in] laddr  Local address information.
          * @param[in] raddr  Remote address information.
          * @param[in] backup Backup priority flag.
+         * @param[in] error  Subflow related event error.
```
small detail, do you mind adding a bit more details, e.g. there might not be any error, it is an `errno`, and the value is a copy of `sk_err`.

Something like that?

    Subflow closing error, if any. The 'errno'
    set in 'sk_err', e.g. reset, timeout, etc.

Do you mind adding that (or something similar) in `include/mptcpd/private/plugin.h` as well please?

matttbe reviewed (changes requested):
Thank you for having looked at that!

I just have two small suggestions if you don't mind.

marco-a-itl committed:
Improve error parameter documentation and test coverage

- Add non-zero test value (110) for test_error_4 to verify different values
- Enhance error parameter documentation to clarify it's an errno from sk_err
- Update both public and private plugin headers with detailed description

marco-a-itl commented:
> Thank you for having looked at that!
> 
> I just have two small suggestions if you don't mind.

Thanks for the review, I've addressed both suggestions:
- Changed test_error_4 to 110 (non-zero value) to test different error values
- Enhanced the error parameter documentation in both headers to clarify it's an errno from sk_err

matttbe reviewed (approved):
Thank you for the update! It looks good to me!
